### DISCUSSION TITLE
Include file base path for configuration

### DISCUSSION THREAD

**Discussion by SebastianLeitz** [jaegertracing/jaeger#3503] [Q&A] [answered]
A number of CLI configuration directives allows to include external files with data into the configuration. 

Example: `--es.tags-as-fields.config-file`

I have achieved the desired result with absolute file paths in my configuration, such as

`es.tags-as-fields.config-file: /the/place/where/jaeger/collector/lives/conf/uniqueTags.txt`

But absolute paths are not really desirable for deployment and operation. I would like to specify a relative path, but so far I failed. I placed that file next to the Jaeger binary (which is in separate directory in my setup), next to the config yaml file (which also has a separate directory) and in their common base directory. But I can't find the correct way to reference the file then (just `uniqueTags.txt` does not work in either case). Does anybody have a hint on what the base directory is that Jaeger uses to reference this external file?

**Comment by yurishkuro** [comment]
If you're running docker image of Jaeger then you can mount relative paths as absolute within the container, so I don't think it's an issue and is a preferable form over knowing where the binary is within the container. 

If you're using a bare metal binary you're in control its working directory.

**Comment by SebastianLeitz** [comment]
Thanks for the fast answer! What does Jaeger treat as its working directory? The parent directory of the binary? The directory the binary was launched from? Or does it come from some configuration directive itself?

**Comment by yurishkuro** [comment, marked as answer]
If you're talking about bare metal binary, the working dir is inherited from the shell that started the binary, i.e. it is whatever you want it to be.
```shell
# config.json is read from the current dir, but Jaeger binary from elsewhere
$ SPAN_STORAGE_TYPE=memory ./cmd/query/query-darwin-amd64 --query.ui-config config.json
```

**Comment by SebastianLeitz** [comment]
That clarifies it very well. Thank you!
